homeassistant/components/essent/sensor.py

```diff
@@ -0,0 +1,280 @@
+    if (data := entity.energy_data) is None:
+        return []
+    return [*data.tariffs, *data.tariffs_tomorrow]
+
+
+def _get_current_tariff(entity: "EssentSensor") -> Tariff | None:
+    """Return the currently active tariff."""
+    now = dt_util.now()
+    for tariff in _get_all_tariffs(entity):
+        start, end = _parse_tariff_times(tariff)
+        if start and end and start <= now < end:
+            return tariff
+    return None
+
+
+def _get_next_tariff(entity: "EssentSensor") -> Tariff | None:
+    """Return the next tariff."""
+    now = dt_util.now()
+    for tariff in _get_all_tariffs(entity):
+        start, _ = _parse_tariff_times(tariff)
+        if start and start > now:
+            return tariff
+    return None
+
+
+def _current_price_attrs(entity: "EssentSensor") -> dict[str, Any]:
+    """Build attributes for the current price."""
+    if (tariff := _get_current_tariff(entity)) is None:
+        return {}
+    groups = {
+        group["type"]: group.get("amount") for group in tariff.groups if "type" in group
+    }
+    return {
+        "price_ex_vat": tariff.total_amount_ex,
+        "vat": tariff.total_amount_vat,
+        "market_price": groups.get("MARKET_PRICE"),
+        "purchasing_fee": groups.get("PURCHASING_FEE"),
+        "tax": groups.get("TAX"),
+        "start_time": _format_dt_str(tariff.start),
+        "end_time": _format_dt_str(tariff.end),
+    }
+
+
+def _next_price_attrs(entity: "EssentSensor") -> dict[str, Any]:
+    """Build attributes for the next price."""
+    if (tariff := _get_next_tariff(entity)) is None:
+        return {}
+    groups = {
+        group["type"]: group.get("amount") for group in tariff.groups if "type" in group
+    }
+    return {
+        "price_ex_vat": tariff.total_amount_ex,
+        "vat": tariff.total_amount_vat,
+        "market_price": groups.get("MARKET_PRICE"),
+        "purchasing_fee": groups.get("PURCHASING_FEE"),
+        "tax": groups.get("TAX"),
+        "start_time": _format_dt_str(tariff.start),
+        "end_time": _format_dt_str(tariff.end),
+    }
+
+
+def _average_attrs(entity: "EssentSensor") -> dict[str, Any]:
+    """Build attributes for the average price."""
+    if (data := entity.energy_data) is None:
+        return {}
+    return {
+        "min_price": data.min_price,
+        "max_price": data.max_price,
+    }
+
+
+def _extreme_attrs(entity: "EssentSensor", target: float) -> dict[str, Any]:
+    """Build attributes for lowest/highest prices."""
+    if (data := entity.energy_data) is None:
+        return {}
+    for tariff in data.tariffs:
+        if tariff.total_amount == target:
+            return {
+                "start": _format_dt_str(tariff.start),
+                "end": _format_dt_str(tariff.end),
+            }
+    return {}
+
+
+SENSORS: tuple[EssentSensorEntityDescription, ...] = (
+    EssentSensorEntityDescription(
+        key="current_price",
+        name="current price",
```

Comment:
If you have the name defined in the strings.json, you can remove the name. If you find that didn't work. You probably didn't generate translations. I'm on mobile so I don't have the command with me right now.

Reply:
Done.